app/models/story.rb

```diff
@@ -229,10 +230,16 @@ def self.find_similar_by_url(url)
     end
     urls = urls2.uniq
 
+    # trailing pound
+    urls.each do |u|
+      urls_with_trailing_pound.push u + "#"
```

Comment:
Is this just for the case where the url already in the database is `https://example.com/foo#`, right? If so, you can remove it - we only have 29 of them. Let's just write a migration to remove them.

Reply:
No, this is instead to take care of the scenario described below. It does so by appending a `#` to the URL and searching for a pattern where it matches the URL plus any characters after the `#`, thus the ugly and expensive `RLIKE`.

Alice submits the following URL https://artagnon.com/articles/leancoq#main, three minutes later Bob tries submitting https://artagnon.com/articles/leanco